### PR TITLE
Remove/detect star imports in repo

### PR DESCRIPTION
**Pull request by avgAGB**

### CODE REVIEW

**Review by cogross — DISMISSED** [review]

**Review by hlgp — DISMISSED** [review]

**Review by jwomeara — CHANGES_REQUESTED** [review]
`junit.framework.*` are JUnit 3.x-style asserts.  Do we not want to use `org.junit.*` for JUnit 4.x-style asserts?

**Comment by avgAGB** [comment]
> `junit.framework.*` are JUnit 3.x-style asserts. Do we not want to use `org.junit.*` for JUnit 4.x-style asserts?

I'll add it in for the next commit

**Review by mineralntl — DISMISSED** [review]

**Review by cogross — DISMISSED** [review]

File: warehouse/accumulo-extensions/src/test/java/datawave/ingest/table/volumeChoosers/ShardedTableDateBasedTieredVolumeChooserTest.java [hunk 1]
```diff
@@ -1,8 +1,7 @@
 package datawave.ingest.table.volumeChoosers;
 
-import static org.junit.Assert.*;
+import static junit.framework.TestCase.assertTrue;
```
Comment (apmoriarty) [inline warehouse/accumulo-extensions/src/test/java/datawave/ingest/table/volumeChoosers/ShardedTableDateBasedTieredVolumeChooserTest.java]: `junit.framework` is deprecated. Looks like intellij defaulted to the wrong import.

**Review by apmoriarty — CHANGES_REQUESTED** [review]

File: warehouse/accumulo-extensions/src/test/java/datawave/ingest/table/volumeChoosers/ShardedTableDateBasedTieredVolumeChooserTest.java (hunk 1 above)
Comment (avgAGB) [reply warehouse/accumulo-extensions/src/test/java/datawave/ingest/table/volumeChoosers/ShardedTableDateBasedTieredVolumeChooserTest.java]: Thanks, will fix.

**Review by MiguelRicardos — DISMISSED** [review]
LGTM

**Comment by matthpeterson** [comment]
> `junit.framework.*` are JUnit 3.x-style asserts. Do we not want to use `org.junit.*` for JUnit 4.x-style asserts?

I interpreted this comment to mean that we should not have `org.junit.*` as a bannedImport because we use it.  I see it's still there so I must be misunderstanding.  Can you help clear this up for me?

I'm also confused how the automated checks are passing.  Maybe this plugin isn't getting executed by them yet.

File: pom.xml [hunk 2]
```diff
@@ -1661,6 +1661,41 @@
...
+                                <phase>process-sources</phase>
+                                <configuration>
+                                    <rules>
+                                        <RestrictImports>
+                                            <!-- Define an explanatory reason why these imports are prohibited -->
+                                            <reason>Remove JUnit Framework</reason>
+                                            <!-- Specify a single pattern to be banned -->
+                                            <bannedImport>junit.framework.*</bannedImport>
+                                            <bannedImport>org.junit.*</bannedImport>
+                                            <bannedImport>**.'*'</bannedImport>
```
Comment (matthpeterson) [inline pom.xml:1688]: I think this bannedImport entry will prevent any imports containing a `*`.  Is that correct?  Is it worth having this entry in a separate group such that the `reason` is correct for it?

Comment (avgAGB) [reply pom.xml:1688]: Yep that's correct. I'll space this out to another category

**Comment by avgAGB** [comment]
> > `junit.framework.*` are JUnit 3.x-style asserts. Do we not want to use `org.junit.*` for JUnit 4.x-style asserts?
> 
> I interpreted this comment to mean that we should not have `org.junit.*` as a bannedImport because we use it. I see it's still there so I must be misunderstanding. Can you help clear this up for me?
> 
> I'm also confused how the automated checks are passing. Maybe this plugin isn't getting executed by them yet.

My thinking was across the board we don't want to use starred imports, so I added in patterns to cover both junit3/junit4 package patterns. I don't think we necessarily need to rely on org.junit.* in any unit tests, as I've went ahead and refactored any classes which were using it and switched to single imports as needed. The plugin may not be taking effect for the gitlab flow until the branch is merged into integration.

**Comment by jwomeara** [comment]
> > > `junit.framework.*` are JUnit 3.x-style asserts. Do we not want to use `org.junit.*` for JUnit 4.x-style asserts?
> > 
> > 
> > I interpreted this comment to mean that we should not have `org.junit.*` as a bannedImport because we use it. I see it's still there so I must be misunderstanding. Can you help clear this up for me?
> > I'm also confused how the automated checks are passing. Maybe this plugin isn't getting executed by them yet.
> 
> My thinking was across the board we don't want to use starred imports, so I added in patterns to cover both junit3/junit4 package patterns. I don't think we necessarily need to rely on org.junit.* in any unit tests, as I've went ahead and refactored any classes which were using it and switched to single imports as needed. The plugin may not be taking effect for the gitlab flow until the branch is merged into integration.

My initial comment has been addressed by your updates.  The request was to stop using `junit.framework` in our tests and use the equivalent `org.junit` imports.

**Review by jwomeara — DISMISSED** [review]
i'm happy with this

**Comment by matthpeterson** [comment]
> My thinking was across the board we don't want to use starred imports, so I added in patterns to cover both junit3/junit4 package patterns. I don't think we necessarily need to rely on org.junit.* in any unit tests, as I've went ahead and refactored any classes which were using it and switched to single imports as needed. 

I thought that the `org.junit.*` entry specified that even single imports would be banned.  This is based on the first of the three special literals listed under https://github.com/skuzzle/restrict-imports-enforcer-rule#package-patterns, copied below.  It looks like the `'*'` entry is what will look for a literal `*` within the import.

> Package patterns are dot separated strings that can be compared case sensitively part by part. Every part must adhere to the java identifier rules with the exception of a some special literals:
> \* matches every package part but exactly one.
> ** matches multiple package parts but at least one.
> '*' matches a literal * in an import statement.

**Comment by avgAGB** [comment]
> > My thinking was across the board we don't want to use starred imports, so I added in patterns to cover both junit3/junit4 package patterns. I don't think we necessarily need to rely on org.junit.* in any unit tests, as I've went ahead and refactored any classes which were using it and switched to single imports as needed.
> 
> I thought that the `org.junit.*` entry specified that even single imports would be banned. This is based on the first of the three special literals listed under https://github.com/skuzzle/restrict-imports-enforcer-rule#package-patterns, copied below. It looks like the `'*'` entry is what will look for a literal `*` within the import.
> 
> > Package patterns are dot separated strings that can be compared case sensitively part by part. Every part must adhere to the java identifier rules with the exception of a some special literals:
> > * matches every package part but exactly one.
> > ** matches multiple package parts but at least one.
> > '*' matches a literal * in an import statement.

Ah, makes sense then. Yeah the semantics on the banned import regexes is a bit confusing, but the '*' in this case is literal. If I wanted to cover everything underneath a package then the '**' would be used.  I updated the latest commit to reflect this for the junit imports.

**Review by matthpeterson — APPROVED** [review]

**Review by apmoriarty — APPROVED** [review]

**Pull request merged**